django_project/realtime/tasks/ash.py

```diff
@@ -119,14 +120,19 @@ def generate_event_report(ash_event):
             task_status=result.state)
 
     # Check impact layer
-    elif not ash_event.impact_layer_exists and ash_event.need_run_analysis:
+    elif (ash_event.hazard_layer_exists and
```

Comment:
Just for a note from our discussion.
I think the conditional should just check their respective task status only and report the errors in their separate task status.

I will experiment on this in earthquake.